test/integration/puppetlabs/puppetserver/testutils.clj

```diff
@@ -125,6 +126,30 @@
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;;; Interacting with puppet code and catalogs
 
+(schema/defn ^:always-validate module-dir
```

Comment:
Yup, it could. Is it typical in this code base to always include a return type schema? I usually find the input types most useful for debugging purposes, return types on simple functions less so